### ISSUE TITLE
Problem with eos_token_id not being used when add_eos_token=True is used

### ISSUE THREAD

**Issue by iliasmiraoui** [huggingface/transformers#26743]
### System Info

transformers==4.35.dev0

I have been playing around with the new Mistral models/finetunes. For training, I believe it is needed to add "add_eos_token=True" on the tokenizer thats being passed to the SFTrainer so that the model still predicts eos_token at the end of the sequence and it's better than including it in the "text" column. I didn't see a lot of docs on templates so want to check I am not inventing this right?

I have noticed that the behavior works well on the regular Mistral model but is completely off on some other models like the OpenOrca fine-tune. While the eos_token exists in the tokenizer and is correctly categorized for the prompt_template, the tokenizer always uses the unknown token instead when tokenizing and "add_eos_token=True" (see screenshot below) which is definitely not right and leads to problems with never-ending generation after finetunes.

<img width="889" alt="Screenshot 2023-10-11 at 9 28 22 PM" src="https://github.com/huggingface/transformers/assets/83364633/b43a774f-b652-4654-ade3-92dc890660d4">

### Who can help?

@ArthurZucker @younesbelkada 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

model_name = "Open-Orca/Mistral-7B-OpenOrca"
tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True,add_eos_token=True)
tokenizer("Test")

### Expected behavior

It should output `{'input_ids': [1, 3735, 32000], 'attention_mask': [1, 1, 1]} `instead of` {'input_ids': [1, 3735, 0], 'attention_mask': [1, 1, 1]}` since `tokenizer.eos_token_id` is 32000.

**Comment by ArthurZucker** [comment]
That is most probably because `normalized = True` for this added token. Thus the actual token that is known to the underlying fast tokenizer is `'▁<|im_end|>'` which is why the `<|im_end|>`  is not recognized and encoded as an unk token.

**Comment by iliasmiraoui** [comment]
I think it was fixed !!
